opm/models/utils/simulator.hh

```diff
@@ -129,25 +129,27 @@ public:
             OPM_END_PARALLEL_TRY_CATCH("Allocating the simulation vanguard failed: ", comm);
         }
 
-        // Only relevant for CpGrid
         if (verbose_) {
-            std::cout << "Adding LGRs, if any\n" << std::flush;
+            std::cout << "Distributing the vanguard's data\n" << std::flush;
         }
 
         {
             OPM_BEGIN_PARALLEL_TRY_CATCH();
-            vanguard_->addLgrs();
-            OPM_END_PARALLEL_TRY_CATCH("Adding LGRs to the simulation vanguard failed: ", comm);
+            vanguard_->loadBalance();
+            OPM_END_PARALLEL_TRY_CATCH("Could not distribute the vanguard data: ", comm);
         }
 
+        // Only relevant for CpGrid
         if (verbose_) {
-            std::cout << "Distributing the vanguard's data\n" << std::flush;
+            std::cout << "Adding LGRs, if any, in serial run\n" << std::flush;
         }
 
         {
-            OPM_BEGIN_PARALLEL_TRY_CATCH();
-            vanguard_->loadBalance();
-            OPM_END_PARALLEL_TRY_CATCH("Could not distribute the vanguard data: ", comm);
+            if ( comm.size() == 1 ) { // serial run
```

Comment:
Should we log a warning if not?

Reply:
Done! Not here though. I moved this if-serial run into CpGridVanguard::addLgrs(), and warn there if it's a parallel run.